### DISCUSSION TITLE
Why do we use appsettings.json and appsettings.Prod.json but not appsettings.Dev.json?

### DISCUSSION THREAD

**Discussion by VictorioBerra** [Dotnet-Boxed/Templates#478] [Q&A]
The default ASPNET templates have an `appsettings.Development.json` specifically for development. For example: Development database connecting strings, verbose logging and debug sinks for Serilog.

What is the motivation behind why we use only `appsettings.json` and `appsettings.Production.json`?

**Comment by RehanSaeed** [comment]
I am forced to use `appsettings.json` for Development and `appsettings.Production.json` for Production because of Kestrel and the way `launchSettings.json` passes the `applicationUrl`:

# launchSettings.json
```json
"Kestrel": {
      "commandName": "Project",
      "launchBrowser": true,
      "launchUrl": "",
      "applicationUrl": "https://localhost:5001;http://localhost:5000",
      "environmentVariables": {
        "ASPNETCORE_ENVIRONMENT": "Development"
      }
  },
    "Kestrel Production": {
      "commandName": "Project",
      "launchBrowser": true,
      "launchUrl": "",
      "environmentVariables": {
        "ASPNETCORE_ENVIRONMENT": "Production"
    }
},
```

The ports specified above conflict and get confused with the ones specified in the file below:

# appsettings.Production.json

```json
  "Kestrel": {
    "Endpoints": {
      "Https": {
        "Url": "https://*:443",
        "Certificate": {
          // TODO
        }
      },
      "Http": {
        "Url": "http://*:80"
      }
  }
},
```

If you can submit a PR to swap these two (so that `appsettings.json` is for Production and `appsettings.Development.json` is for Development), while still getting the launch configurations working i.e. the app starts all all URL's specified in `applicationUrl`, I'd merge it in a second.

**Comment by VictorioBerra** [comment]
Great work on this @RehanSaeed!
